pyuoi/linear_model/lasso.py

```
@@ -1,9 +1,78 @@
+import pycasso
+import numpy as np
+from warnings import warn
+
 from sklearn.linear_model import Lasso, LinearRegression
 from sklearn.linear_model.coordinate_descent import _alpha_grid
-
 from .base import AbstractUoILinearRegressor
 
 
+class PycLasso():
+    '''class PycLasso: Lasso using the pycasso solver. Solves for an
+    entire regularization path at once.
+
+        alphas : ndarray
+            regularization path. Defaults to None for compatibility with UoI,
+            but needs to be set prior to fitting
+
+        fit_intercept : bool
+            Should we fit an intercept?
+
+        max_iter : int
+            Iterations for pycasso solver
+
+    '''
+    def __init__(self, alphas=None, fit_intercept=False, max_iter=1000):
+        self.max_iter = max_iter
+        self.fit_intercept = fit_intercept
+        self.alphas = alphas
+
+        # Flag to prevent us from predicting before fitting
+        self.isfitted = False
+
+    def init_solver(self, X, y, alphas):
+
+        self.solver = pycasso.Solver(X, y, family='gaussian',
+                                     useintercept=self.fit_intercept,
+                                     lambdas=alphas,
+                                     penalty='l1',
+                                     max_ite=self.max_iter)
+
+    def set_params(self, **kwargs):
+
+        _valid_params = ['alphas', 'fit_intercept', 'max_iter']
+
+        for key, value in kwargs.items():
+            if key in _valid_params:
+                setattr(self, key, value)
+            else:
+                ValueError('Invalid parameter %s' % key)
+
+    def predict(self, X):
+
+        if self.isfitted:
+
+            return np.matmul(X, self.coef_.T) + self.intercept_
+
+        else:
+
+            warn('Cannot predict, estimator is not yet fit!')
+
+    def fit(self, X, y):
+
+        if self.alphas is None:
+            print('Set alphas before fitting!')
```

This should raise an error.